Update issue templates

ulupo opened the pull request:
Addresses and potentially fixes #349.

ulupo committed:
Update issue templates

Addresses and potentially fixes #349.

ulupo committed:
Delete ISSUE_TEMPLATE.md

ulupo committed:
Update formatting to agree with new issue templates

wreise commented on .github/ISSUE_TEMPLATE/feature_request.md:
```diff
@@ -0,0 +1,30 @@
+**Describe the solution you'd like**
+<!--
+A clear and concise description of what you want to happen.
+-->
```
What do you think about encouraging users to provide (pseudo-)code, for API changes they would like to see?

ulupo replied:
I'll update the description to say that (pseudo-)code is welcome. But users who would not even be sure where to begin should not be discouraged either.

wreise commented on PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,22 +1,19 @@
 <!--
 Thanks for contributing a pull request! Please ensure you have taken a look at
-the contribution guidelines: https://github.com/giotto-ai/giotto-tda/blob/master/CONTRIBUTING.md#pull-request-checklist
+the contribution guidelines: https://giotto-ai.github.io/gtda-docs/dev/contributing/#guidelines
```
amazing, slowly migrating. Do you want to point to dev, or maybe nightly? I think we discussed that the 'default' version would be nightly, but maybe for contributors, you want to point them to dev?

ulupo replied:
Yes, I'd think dev is the more relevant one for contributors.

wreise reviewed (approved):
The current PR template is  ok, but maybe it is an opportunity to  discuss incorporating steps like a checklist, or type of changes? I looked at  some templates, and maybe we could steal some ideas from, for example, [this one](https://gist.github.com/wreise/2af40944dd7c1f23c679fae938e9e79c#file-pull_request_template-md)?
( found at https://www.talater.com/open-source-templates/#/page/99)

ulupo committed:
Wording improvement in feature request template

ulupo committed:
Improve PR template with @wreise's suggestions

- Add a "types of changes" checklist
- Add a checklist to enhance awareness of important steps

Thanks @wreise for these suggestions
